azkaban-common/src/main/java/azkaban/executor/ExecutableNode.java

```diff
@@ -324,6 +330,8 @@ public void fillExecutableFromMapObject(
     this.id = wrappedMap.getString(ID_PARAM);
     this.type = wrappedMap.getString(TYPE_PARAM);
     this.condition = wrappedMap.getString(CONDITION_PARAM);
+    this.conditionOnJobStatus = ConditionOnJobStatus.fromString(wrappedMap.getString
```

Comment:
What if the condition is null? Should we handle the edge case?

Reply:
The condition can be null. If it's null, it just follows our current flow execution logic. 
However the conditionOnJobStatus is `all_success` by default, so it won't be null. If no condition is specified, by default we just follow the current flow execution logic which is `all_success`.